### PR TITLE
AST-882 Fix ARM GCC-12 build

### LINKED ISSUE #1193: Build error on Arm (aarch64, armv7, armv6) -  `error: narrowing conversion of '255' from 'int' to 'char' [-Wnarrowing]`
On openSUSE Tumbleweed, casacore 3.4.0 fails to build on Arm (aarch64, armv7, armv6) with GCC12.

Error log:
```
[ 1730s] [1696/1846] /usr/bin/g++ -DBOOST_PYTHON_DYN_LINK -DBOOST_PYTHON_NO_LIB -DCFITSIO_VERSION_MAJOR=4 -DCFITSIO_VERSION_MINOR=100 -DHAVE_FFTW3 -DHAVE_FFTW3_THREADS -DHAVE_HDF5 -DHAVE_READLINE -DUSE_THREADS -DWCSLIB_VERSION_MAJOR=7 -DWCSLIB_VERSION_MINOR=11 -Dcasa_python3_EXPORTS -I/home/abuild/rpmbuild/BUILD/casacore-3.4.0 -I/home/abuild/rpmbuild/BUILD/casacore-3.4.0/build -I/usr/include/cfitsio -I/usr/lib64/python3.8/site-packages/numpy/core/include -I/usr/include/python3.8 -fcx-fortran-rules -mbranch-protection=standard -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG -fsigned-char -std=c++11 -fopenmp -pthread -O2 -g -DNDEBUG -fPIC -MD -MT python3/CMakeFiles/casa_python3.dir/__/python/Converters/PycArray.cc.o -MF python3/CMakeFiles/casa_python3.dir/__/python/Converters/PycArray.cc.o.d -o python3/CMakeFiles/casa_python3.dir/__/python/Converters/PycArray.cc.o -c /home/abuild/rpmbuild/BUILD/casacore-3.4.0/python/Converters/PycArray.cc
[ 1730s] FAILED: python3/CMakeFiles/casa_python3.dir/__/python/Converters/PycArray.cc.o 
[ 1730s] /usr/bin/g++ -DBOOST_PYTHON_DYN_LINK -DBOOST_PYTHON_NO_LIB -DCFITSIO_VERSION_MAJOR=4 -DCFITSIO_VERSION_MINOR=100 -DHAVE_FFTW3 -DHAVE_FFTW3_THREADS -DHAVE_HDF5 -DHAVE_READLINE -DUSE_THREADS -DWCSLIB_VERSION_MAJOR=7 -DWCSLIB_VERSION_MINOR=11 -Dcasa_python3_EXPORTS -I/home/abuild/rpmbuild/BUILD/casacore-3.4.0 -I/home/abuild/rpmbuild/BUILD/casacore-3.4.0/build -I/usr/include/cfitsio -I/usr/lib64/python3.8/site-packages/numpy/core/include -I/usr/include/python3.8 -fcx-fortran-rules -mbranch-protection=standard -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG -fsigned-char -std=c++11 -fopenmp -pthread -O2 -g -DNDEBUG -fPIC -MD -MT python3/CMakeFiles/casa_python3.dir/__/python/Converters/PycArray.cc.o -MF python3/CMakeFiles/casa_python3.dir/__/python/Converters/PycArray.cc.o.d -o python3/CMakeFiles/casa_python3.dir/__/python/Converters/PycArray.cc.o -c /home/abuild/rpmbuild/BUILD/casacore-3.4.0/python/Converters/PycArray.cc
[ 1730s] In file included from /usr/include/boost/integer.hpp:20,
[ 1730s]                  from /usr/include/boost/function/function_base.hpp:20,
[ 1730s]                  from /usr/include/boost/function/detail/prologue.hpp:17,
[ 1730s]                  from /usr/include/boost/function/function_template.hpp:13,
[ 1730s]                  from /usr/include/boost/function/detail/maybe_include.hpp:15,
[ 1730s]                  from /usr/include/boost/function/function0.hpp:11,
[ 1730s]                  from /usr/include/boost/python/errors.hpp:13,
[ 1730s]                  from /usr/include/boost/python/handle.hpp:11,
[ 1730s]                  from /usr/include/boost/python/args_fwd.hpp:10,
[ 1730s]                  from /usr/include/boost/python/args.hpp:10,
[ 1730s]                  from /usr/include/boost/python.hpp:11,
[ 1730s]                  from /home/abuild/rpmbuild/BUILD/casacore-3.4.0/casacore/python/Converters/PycArray.h:34,
[ 1730s]                  from /home/abuild/rpmbuild/BUILD/casacore-3.4.0/casacore/python/Converters/PycArray.tcc:31,
[ 1730s]                  from /home/abuild/rpmbuild/BUILD/casacore-3.4.0/python/Converters/PycArray.cc:30:
[ 1730s] /usr/include/boost/integer_traits.hpp:83:64: error: narrowing conversion of '255' from 'int' to 'char' [-Wnarrowing]
[ 1730s]    83 |     public detail::integer_traits_base<char, CHAR_MIN, CHAR_MAX>
[ 1730s]       |                                                                ^
```

### PR DESCRIPTION
**Pull request by mnijhuis-tos**
Support systems where 'char' is unsigned.

Should fix #1193

### CODE REVIEW

**Comment by aroffringa** [comment]
Nice!

**Review by aroffringa — APPROVED** [review]

**Comment by mnijhuis-tos** [comment]
> Do we know why this was added and are we sure this change doesn't break things?
Without context, it's difficult to understand what you are referring to with the first "this".

File: CMakeLists.txt [hunk 1]
```diff
@@ -185,7 +185,7 @@ if (NOT CMAKE_CXX_FLAGS)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unneeded-internal-declaration")
     endif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
 endif (NOT CMAKE_CXX_FLAGS)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsigned-char")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -funsigned-char")
```
Comment (mnijhuis-tos) [inline CMakeLists.txt:188]: This change is only for checking if the tests pass with unsigned characters. I'll remove it before merging.

Comment (tammojan) [reply CMakeLists.txt]: Since this change is gone, is this PR now out of the 'draft' state?

**Review by aroffringa — APPROVED** [review]

**Pull request merged**